HLshell/mycmd_1.py

```diff
@@ -334,57 +326,175 @@ def get_names(self): ## Don't change the name
-            if (len(paramList)==2): print('Second parameters (port) should be a number.\nUse default setting, port ', self._DEFAULTPORT())
-
+            if (len(paramList)==2): 
+                print_err('Second parameters (port) should be a number.\nUse default setting, port ', self._DEFAULTPORT())
+        
+        connect_timeout = clientShell.DEFAULT_TIMEOUT
+        self.socket = socket.socket()
+        self.socket.settimeout(connect_timeout)
+        print('Attempting to connect to ' + str(self.hostIP) + ':' + str(self.port))
         try:
             self.socket.connect((self.hostIP,self.port))
         except socket.error as msg:
-            print(msg,file=sys.stderr)
-            return
+            print_err('Server is not receiving connections')
+            self.is_connecting = False
+            return 
+        except TypeError:
+            print_err('Please enter a proper IP')
+            self.is_connecting = False
+            return 
+        except socket.timeout:
+            print_err('Connection timed out after ' + str(connect_timeout) + ' seconds')
+            self.is_connecting = False
+            return 
 
+        self.socket.settimeout(shell.DEFAULT_TIMEOUT)
         self.socket.sendall(socket.gethostname().encode())
-        cwd = self.socket.recv(1024)
-        print("STATUS: connect to machine: ", cwd.decode())
-        self.prompt = 'FTScmd> '
-
-
-    def do_wait(self,par):
-        '''press CTRL+C to stop receiving'''
-        print('Waiting for the command...')
-        print('press CTRL+C to stop receiving')
+        
         try:
-            while True:
-                command = self.socket.recv(self.BUFFER_SIZE).decode()
-                print('Received command: ', command)
+            cwd = self.socket.recv(1024)
+            print("STATUS: connect to machine: ", cwd.decode())
+            self.prompt = 'FTScmd> '
+            self.is_connected = True
+        except (ConnectionAbortedError,ConnectionResetError) as e :
+            print_err('Connection not ready, try again')
+            self.is_connecting = False
+            return 
+        except socket.timeout:
+            print_err('Connection timed out after ' + str(connect_timeout) + ' seconds')
+            self.is_connecting = False
+            return 
+        self.is_connecting = False
+
+    def receive_command(self, timeout, EXIT='q'):
+        '''
+        Waits for a command from the server every [timeout] seconds
+        If connection is lost in the middle of receiving a command,
+        then we exit the loop
+        '''
+        try:
+            self.socket.settimeout(timeout)
+            while(self.waiting_for_cmd):
+                try:
+                    command = self.socket.recv(shell.BUFFER_SIZE).decode()
+                except socket.timeout:
+                    continue 
+                print('Command received: ' + str(command))
+                cmd_received = pickle.dumps(['Command Received', None])
```

Comment:
pickple lib is used to pack list to send or you have to send n times if you have n lines.
For this case, it's not necessary

Reply:
I just wanted to be consistent.